src/components/KDateTimePicker/KDateTimePicker.vue

```diff
@@ -0,0 +1,838 @@
+    },
+    /**
+     * Upper bound for `v-calendar` dates, everything after this date will be disabled
+     */
+    maxDate: {
+      type: Date,
+      required: false,
+      default: null,
+    },
+    /**
+     * Lower bound for `v-calendar` dates, everything preceding this date will be disabled
+     */
+    minDate: {
+      type: Date,
+      required: false,
+      default: null,
+    },
+    /**
+     * Sets a custom interval for the Minute select dropdown
+     */
+    minuteIncrement: {
+      type: Number,
+      required: false,
+      default: 5,
+    },
+    /**
+     * Determines which `v-calendar` type to initialize.
+     * Three of the values (`date`, `time`, `dateTime`) are passed verbatim to `v-calendar`,
+     * whereas `relative` denotes a component instance made up solely of time frames.
+     */
+    mode: {
+      type: String,
+      required: true,
+      validator: (value: string) => {
+        return ['relative', 'date', 'time', 'dateTime'].includes(value)
+      },
+    },
+    /**
+     * Help text displayed as the default mesage inside the input field.
+     * When "Clear" is clicked, the input will revert to displaying this.
+     */
+    placeholder: {
+      type: String,
+      required: false,
+      default: 'Select a time range',
+    },
+    /**
+     * Determines whether the `v-calendar` will allow a single date/time,
+     * or a range of dates/times.
+     */
+    range: {
+      type: Boolean,
+      required: false,
+      default: false,
+    },
+    /**
+     * A custom set of time frames to be displayed as selectable buttons.
+     * The `timeframeLength`, `start`, and `end` values are passed in as functions,
+     * allowing for on-the-fly date boundary creation.
+     */
+    timePeriods: {
+      type: Array as PropType<Array<TimeFrameSection>>,
+      required: false,
+      default: () => [],
+      validator: (sectionsArray: TimeFrameSection[]) => {
+        return sectionsArray.every((item: TimeFrameSection) => {
+          return Array.isArray(item.values) && item.values.every((timeframe: TimePeriod) => {
+            // Check validity of each timeframe
+            return typeof timeframe.timeframeText === 'string' &&
+              timeframe.timeframeLength !== undefined &&
+              typeof timeframe.key === 'string' &&
+              timeframe.key !== undefined &&
+              typeof timeframe.display === 'string' &&
+              timeframe.display !== undefined &&
+              timeframe.start !== undefined &&
+              timeframe.end !== undefined
+          })
+        })
+      },
+    },
+    /**
+     * Sets the input field to a fixed width
+     */
+    width: {
+      type: String,
+      required: false,
+      default: 'auto',
+    },
+  },
+  emits: ['input', 'change', 'update:modelValue'],
+  setup(props, { emit }) {
+    const modelConfig = { type: 'number' } // https://vcalendar.io/datepicker.html#model-config
```

Comment:
nit: can we move the comment above the line?
```suggestion
    // https://vcalendar.io/datepicker.html#model-config
    const modelConfig = { type: 'number' }
```